### ISSUE TITLE
download from iCloud Drive

### ISSUE THREAD

**Issue by hlgrftschn** [cross-platform/icloud-for-linux#42]
Hi, I'm new on GitHub. 
thanks for your great work
iCloud works for me, but I cant download files from iCloud Drive and from Fotos.
Upload works with both and no, there are no downloaded files in ~/downloads
Ah and sorry, I use Linux Mint 20 and snap 2.50.1 with icloud-for-linux 0.11

**Comment by MarcusTomlinson** [comment]
Could you run Drive from the terminal like so: `icloud-for-linux.drive` then try to download a file and share the output from the terminal here. Thanks!

**Comment by hlgrftschn** [comment]
Thanks Marcus,
I can't download any file, but I got this output when I do `icloud-for-linux.drive ` 
```
(base) holger@minti:~$ icloud-for-linux.drive
Qt: Session management error: Could not open network socket
Sandboxing disabled by user.
js: Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.
js: Unknown key isCKMail found in infoJSON
js: Unknown key isCKMail found in infoJSON
js: Unknown key isCKMail found in infoJSON
js: Unknown key isCKMail found in infoJSON
js: Unknown key isCKMail found in infoJSON
js: Unknown key isCKMail found in infoJSON
js: Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.
(base) holger@minti:~$
```
Should I enable Sandboxing? How?

Thanks Holger

**Comment by MarcusTomlinson** [comment]
> Should I enable Sandboxing? How?

Nah, it won't have anything to do with that. Sandboxing is disabled app-level as the snap naturally takes care of that.

Could you try:

```
rm -rf ~/snap/icloud-for-linux/
icloud-for-linux.drive
```

(sorry, you will then have to log in again)

**Comment by hlgrftschn** [comment]
Sorry for the late response.
It doesn't work :-(
```
(base) holger@minti:~$ icloud-for-linux.drive
Qt: Session management error: Could not open network socket
Sandboxing disabled by user.
js: Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.
js: Unknown key isCKMail found in infoJSON
js: Unknown key isCKMail found in infoJSON
js: Unknown key isCKMail found in infoJSON
js: Unknown key isCKMail found in infoJSON
js: Unknown key isCKMail found in infoJSON
js: Unknown key isCKMail found in infoJSON

```
I can create folder, delete folder and files but I can't download any files.
Any ideas?

Regards Holger

**Comment by Monikae** [comment]
> Could you run Drive from the terminal like so: `icloud-for-linux.drive` then try to download a file and share the output from the terminal here. Thanks!

This results in the following console output on Kubuntu 20.04:

[8228:8333:0529/170617.441422:ERROR:object_proxy.cc(616)] Failed to call method: org.freedesktop.PowerManagement.Inhibit.Inhibit: object_path= /org/freedesktop/PowerManagement/Inhibit: org.freedesktop.DBus.Error.AccessDenied: An AppArmor policy prevents this sender from sending this message to this recipient; type="method_call", sender=":1.96" (uid=1000 pid=8228 comm="icloud-for-linux iclouddrive Drive " label="snap.icloud-for-linux.drive (enforce)") interface="org.freedesktop.PowerManagement.Inhibit" member="Inhibit" error name="(unset)" requested_reply="0" destination="org.freedesktop.PowerManagement" (uid=1000 pid=1825 comm="/usr/lib/x86_64-linux-gnu/libexec/org_kde_powerdev" label="unconfined")
[8228:8333:0529/170617.441867:ERROR:power_save_blocker_x11.cc(335)] No response to Inhibit() request!
[8228:8333:0529/170617.450821:ERROR:object_proxy.cc(616)] Failed to call method: org.freedesktop.PowerManagement.Inhibit.UnInhibit: object_path= /org/freedesktop/PowerManagement/Inhibit: org.freedesktop.DBus.Error.AccessDenied: An AppArmor policy prevents this sender from sending this message to this recipient; type="method_call", sender=":1.96" (uid=1000 pid=8228 comm="icloud-for-linux iclouddrive Drive " label="snap.icloud-for-linux.drive (enforce)") interface="org.freedesktop.PowerManagement.Inhibit" member="UnInhibit" error name="(unset)" requested_reply="0" destination="org.freedesktop.PowerManagement" (uid=1000 pid=1825 comm="/usr/lib/x86_64-linux-gnu/libexec/org_kde_powerdev" label="unconfined")
[8228:8333:0529/170617.450859:ERROR:power_save_blocker_x11.cc(408)] No response to Uninhibit() request!

**Comment by MarcusTomlinson** [comment]
This should be fixed in v0.12 now. Please run `snap refresh icloud-for-linux`.
